plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -84,27 +85,26 @@ public static IcebergTableHandle fromJsonForDeserializationOnly(
             @JsonProperty("retryMode") RetryMode retryMode,
             @JsonProperty("updatedColumns") List<IcebergColumnHandle> updatedColumns)
     {
-        return new IcebergTableHandle(
-                schemaName,
-                tableName,
-                tableType,
-                snapshotId,
-                tableSchemaJson,
-                partitionSpecJson,
-                formatVersion,
-                unenforcedPredicate,
-                enforcedPredicate,
-                projectedColumns,
-                nameMappingJson,
-                tableLocation,
-                storageProperties,
-                retryMode,
-                updatedColumns,
-                false,
-                Optional.empty());
+        return builder()
+                .withSchemaName(schemaName)
```

Comment:
I would expect any required parameters to be in the constructor of the builder. ie:

```
new Builder(tableLocation, ...)
  .withOptionalParameter(opt1)
  ...;
```

Reply:
We have many builders in Trino, this can be said on many of them.
IMO this is the developer's responsibility. The developer could also pass `null` to the constructor.

CC @raunaqmorarka

Reply:
@assaf2 let's update it to make the required parameters part of builder constructor.

Reply:
@raunaqmorarka @alexjo2144 which are the actually required ones for Hive+Iceberg? Most of the params are wrapped with `requireNonNull`.
In addition, I can make the constructors protected to support cases in which the table handles are inherited.

Reply:
Since there are no derived classes of these table handles in this repo, we shouldn't make the constructor protected.
I think the required parameters for HiveTableHandle are schemaName, tableName and transaction.
For IcebergTableHandle it seems to be schemaName, tableName, tableType, tableSchemaJson, formatVersion, tableLocation.

Reply:
@raunaqmorarka can't we use `NO_ACID_TRANSACTION` as a default for `transaction`?

Reply:
yes, that seems reasonable